llvm/lib/Target/X86/X86AvoidStoreForwardingBlocks.cpp

```diff
@@ -289,10 +289,8 @@ static unsigned getYMMtoXMMStoreOpcode(unsigned StoreOpcode) {
 }
 
 static int getAddrOffset(const MachineInstr *MI) {
-  const MCInstrDesc &Descl = MI->getDesc();
-  int AddrOffset = X86II::getMemoryOperandNo(Descl.TSFlags);
+  const int AddrOffset = X86::getFirstAddrOperandIdx(*MI);
```

Comment:
The `const` doesn't work here. Change the function to `static const int` or remove it.

Reply:
Is it just a stylstic preference? Or why doesn't declaring `AddrOffset` as `const` work here?

Reply:
We usually don't use `const` for integer tpye. I think you may want to highlight `AddrOffset` is the final offset with bias and don't change it anymore by using `const`. In that way, it doesn't work here, because the function doesn't consume `AddrOffset` but return it in a non-const int.

Reply:
Thanks, fixed.